feature/experimental/gribi/ate_tests/route_removal_during_failover_test/route_removal_during_failover_test.go

```diff
@@ -0,0 +1,609 @@
+		sipv4.Enabled = ygot.Bool(true)
+	}
+
+	a := sipv4.GetOrCreateAddress(dutIPv4)
+	a.PrefixLength = ygot.Uint8(uint8(ipv4PrefixLen))
+}
+
+// configureATE configures a single ATE layer 3 interface.
+func configureATE(t *testing.T, top *ondatra.ATETopology, atePort *ondatra.Port, Name string, vlanID uint16, dutIPv4 string, ateIPv4 string) {
+	t.Helper()
+
+	i := top.AddInterface(Name).WithPort(atePort)
+	if vlanID != 0 {
+		i.Ethernet().WithVLANID(vlanID)
+	}
+	i.IPv4().WithAddress(ateIPv4)
+	i.IPv4().WithDefaultGateway(dutIPv4)
+}
+
+// awaitTimeout calls a fluent client Await, adding a timeout to the context.
+func awaitTimeout(ctx context.Context, c *fluent.GRIBIClient, t testing.TB, timeout time.Duration) error {
+	subctx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+	return c.Await(subctx, t)
+}
+
+// testArgs holds the objects needed by a test case.
+type testArgs struct {
+	ctx        context.Context
+	client     *fluent.GRIBIClient
+	dut        *ondatra.DUTDevice
+	ate        *ondatra.ATEDevice
+	top        *ondatra.ATETopology
+	electionID gribi.Uint128
+}
+
+// createTrafficFlow generates traffic flow from source network to
+// destination network via srcEndPoint to dstEndPoint and checks for
+// packet loss.
+func createTrafficFlow(t *testing.T, ate *ondatra.ATEDevice, top *ondatra.ATETopology) *ondatra.Flow {
+	ethHeader := ondatra.NewEthernetHeader()
+	ipv4Header := ondatra.NewIPv4Header()
+	ipv4Header.DstAddressRange().
+		WithMin("198.18.196.1").
+		WithMax("198.18.199.255").
+		WithCount(1020)
+
+	srcEndPoint := top.Interfaces()["src"]
+	dstEndPoint := []ondatra.Endpoint{}
+	for intf, intfData := range top.Interfaces() {
+		if intf != "src" {
+			dstEndPoint = append(dstEndPoint, intfData)
+		}
+	}
+
+	flow := ate.Traffic().NewFlow("Flow").
+		WithSrcEndpoints(srcEndPoint).
+		WithDstEndpoints(dstEndPoint...).
+		WithHeaders(ethHeader, ipv4Header)
+
+	return flow
+}
+
+// Function to send traffic
+func sendTraffic(t *testing.T, ate *ondatra.ATEDevice, flow *ondatra.Flow) {
+	t.Logf("Starting traffic")
+	ate.Traffic().Start(t, flow)
+}
+
+// Function to verify traffic
+func verifyTraffic(t *testing.T, ate *ondatra.ATEDevice, flow *ondatra.Flow) {
+	t.Helper()
+	flowPath := gnmi.OC().Flow(flow.Name())
+	if got := gnmi.Get(t, ate, flowPath.LossPct().State()); got > 0 {
+		t.Errorf("LossPct for flow %s got %g, want 0", flow.Name(), got)
+	} else {
+		t.Logf("Traffic flows fine from ATE-port1 to ATE-port2")
+	}
+}
+
+// findSecondaryController finds out primary and secondary controllers
+func findSecondaryController(t *testing.T, dut *ondatra.DUTDevice, controllers []string) (string, string) {
+	t.Helper()
+	var primary, secondary string
+	for _, controller := range controllers {
+		role := gnmi.Get(t, dut, gnmi.OC().Component(controller).RedundantRole().State())
+		t.Logf("Component(controller).RedundantRole().Get(t): %v, Role: %v", controller, role)
+		if role == secondaryController {
```

Comment:
a little more idiomatic to use a switch statement in Go: https://go.dev/doc/effective_go#switch